packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -265,18 +268,14 @@ export function buildRootObject() {
           details`keyword ${keyword} must be unique`,
         );
       },
-      saveIssuer: (issuerP, keyword) => {
+      saveIssuer: async (issuerP, keyword) => {
         // TODO: The checks of the keyword for uniqueness are
         // duplicated. Assess how waiting on promises to resolve might
         // affect those checks and see if one can be removed.
         zcf.assertUniqueKeyword(keyword);
-        return E(zoeInstanceAdmin)
-          .saveIssuer(issuerP, keyword)
-          .then(() => {
-            return issuerTable
-              .initIssuer(issuerP)
-              .then(record => registerIssuerRecordWithKeyword(keyword, record));
-          });
+        await E(zoeInstanceAdmin).saveIssuer(issuerP, keyword);
+        const record = await issuerTable.initIssuer(issuerP);
```

Comment:
I like the style that calls for comments highlighting the awaits.

Reply:
Ok sounds good